kinesis/src/main/scala/akka/stream/alpakka/kinesis/CommittableRecord.scala

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (C) 2016-2019 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.kinesis
+
+import akka.Done
+import software.amazon.kinesis.lifecycle.ShutdownReason
+import software.amazon.kinesis.retrieval.KinesisClientRecord
+import software.amazon.kinesis.retrieval.kpl.ExtendedSequenceNumber
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class CommittableRecord(
+    val shardId: String,
+    val recordProcessorStartingSequenceNumber: ExtendedSequenceNumber,
+    val millisBehindLatest: Long,
+    val record: KinesisClientRecord,
+    recordProcessor: ShardProcessor,
+    checkpointer: KinesisClientRecord => Unit
+)(implicit executor: ExecutionContext) {
```

Comment:
A bit unexpected to have an execution context within stream elements.

Reply:
Yeah, this looks very ugly. Should I just move it to the `tryToCheckpoint` method of this class?